plugins/source/datadog/resources/plugin/tables.go

```diff
@@ -12,11 +12,47 @@ import (
 	"github.com/cloudquery/cloudquery/plugins/source/datadog/resources/services/slos"
 	"github.com/cloudquery/cloudquery/plugins/source/datadog/resources/services/synthetics"
 	"github.com/cloudquery/cloudquery/plugins/source/datadog/resources/services/users"
-	"github.com/cloudquery/plugin-sdk/v3/schema"
+	"github.com/cloudquery/plugin-sdk/v4/caser"
+	"github.com/cloudquery/plugin-sdk/v4/docs"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
 )
 
-func Tables() []*schema.Table {
-	return []*schema.Table{
+func addCqIDs(table *schema.Table) {
+	havePks := len(table.PrimaryKeys()) > 0
+	cqIdColumn := schema.CqIDColumn
+	if !havePks {
+		cqIdColumn.PrimaryKey = true
+	}
+	table.Columns = append(
+		schema.ColumnList{
+			cqIdColumn,
+			schema.CqParentIDColumn,
+		},
+		table.Columns...,
+	)
+	for _, rel := range table.Relations {
+		addCqIDs(rel)
+	}
+}
+
+func titleTransformer(table *schema.Table) string {
+	if table.Title != "" {
+		return table.Title
+	}
+	exceptions := make(map[string]string)
+	for k, v := range docs.DefaultTitleExceptions {
+		exceptions[k] = v
+	}
+	for k, v := range customExceptions {
+		exceptions[k] = v
+	}
+	csr := caser.New(caser.WithCustomExceptions(exceptions))
+	return csr.ToTitle(table.Name)
+}
```

Comment:
```suggestion
func titleTransformer() func (table *schema.Table) {
	if table.Title != "" {
		return table.Title
	}
	exceptions := make(map[string]string)
	maps.Copy(exceptions, docs.DefaultTitleExceptions)
	maps.Copy(exceptions, customExceptions)
	csr := caser.New(caser.WithCustomExceptions(exceptions))
	return func (table *schema.Table) {
		table.Title =  csr.ToTitle(table.Name)
	}
}
```
 [`maps.Copy`](https://pkg.go.dev/golang.org/x/exp/maps#Copy)